src/common/components/repository-row/index.js

```diff
@@ -135,6 +147,19 @@ class RepositoryRow extends Component {
     let caption;
     if (registerNameStatus.success) {
       caption = <div>{ tickIcon } Registered successfully</div>;
+    } else if ( registerNameStatus.error
+      && registerNameStatus.error.json
+      && registerNameStatus.error.json.payload
+      && registerNameStatus.error.json.payload.code === 'already_registered') {
```

Comment:
Non-blocking comment: We really need to review all error objects in the codebase so that this isn't necessary. If `registerNameStatus.error` is defined, there really should be a standard property with a unique status identifier (such as 'already_registered') on the error object.